test/Tests.Infrastructure/TestBase.cs

```diff
@@ -65,6 +69,16 @@ public void DoNotReuseServer(IDictionary<string, string> customSettings = null)
             _doNotReuseServer = true;
         }
 
+        protected static string GenerateAndSaveSelfSignedCertificate()
```

Comment:
This is racy, on many levels.
If there isn't a `c:\temp` directory, it will fail.
If two threads are running it, it will fail.